tripal_chado/src/Plugin/Field/FieldType/ChadoContactByRoleTypeDefault.php

```diff
@@ -0,0 +1,302 @@
+    //    - description
+    $terms['description'] = $mappingObj->getColumnTermId($object_table, 'description');
+    $max_lengths['description'] = $object_schema_def['fields']['description']['size'];
+    //    - contact type
+    $cvterm_schema_def = $schemaObj->getTableDef('cvterm', ['format' => 'Drupal']);
+    $terms['contact_type'] = $mappingObj->getColumnTermId('cvterm', 'name');
+    $max_lengths['contact_type'] = $cvterm_schema_def['fields']['name']['size'];
+    // C) LINKING TABLE.
+    [$linker_table, $linker_fkey_column] = self::get_linker_table_and_column($storage_settings, $base_table, $object_pkey_col);
+    $linker_schema_def = $schemaObj->getTableDef($linker_table, ['format' => 'Drupal']);
+    //    - primary key
+    $linker_pkey_col = $linker_schema_def['primary key'];
+    $terms['linker_pkey'] = $terms['record_id'];
+    //    - left table foreign key
+    $linker_left_col = array_keys($linker_schema_def['foreign keys'][$base_table]['columns'])[0];
+    $terms['linker_left'] = $mappingObj->getColumnTermId($linker_table, $linker_left_col);
+    //    - right table foreign key
+    $terms['linker_right'] = $mappingObj->getColumnTermId($linker_table, $linker_fkey_column);
+    //    - linking type
+    $terms['linker_type_id'] = $mappingObj->getColumnTermId($linker_table, 'type_id');
+    if (empty($terms['linker_type_id'])) {
+      $terms['linker_type_id'] = $terms['contact_type'];
+    }
+    //    - rank
+    $terms['linker_rank'] = $mappingObj->getColumnTermId($linker_table, 'rank');
+    if (empty($terms['linker_rank'])) {
+      $terms['linker_rank'] = 'OBCS:0000117';
+    }
+
+    // We need to create a table alias for our linker table in order to ensure
+    // contact links with other roles are not combined.
+    // The type used when creating the linking record will be the same as the
+    // type set for the field. As such, we grab that here and use it in our
+    // table alias.
+    $field_settings = $field_definition->getSettings();
+    $term = $field_settings['termIdSpace'] . ':' . $field_settings['termAccession'];
+    $table_alias = $linker_table . '_' . preg_replace( '/[^a-z0-9]+/', '', strtolower( $term ) );
+
+    // FINALLY, THE PROPERTIES!
+    $properties = [];
+
+    // Define the base table record id.
+    $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'record_id', $terms['base_pkey'], [
+      'action' => 'store_id',
+      'drupal_store' => TRUE,
+      'path' => $base_table . '.' . $base_pkey_col,
+    ]);
+
+    // This property will store the Drupal entity ID of the linked chado
+    // record, if one exists.
+    $properties[] = new ChadoIntStoragePropertyType($entity_type_id, self::$id, 'entity_id', self::$drupal_entity_term, [
+      'action' => 'function',
+      'drupal_store' => TRUE,
+      'namespace' => self::$chadostorage_namespace,
+      'function' => self::$drupal_entity_callback,
+      'fkey' => self::$object_id,
```

Comment:
```suggestion
      'function' => self::$drupal_entity_callback,
      'ftable' => self::$object_table,
      'fkey' => self::$object_id,
```
This one line is all that is missing to make the entity links work!